src/collected_output.rs

```diff
@@ -30,8 +30,8 @@ impl Waiter {
         });
         let mut context_clone = context.clone();
         let relay_stdout = config.relay_stdout;
-        let stdout_join_handle = thread::spawn(move || -> io::Result<Vec<u8>> {
-            let mut collected_stdout = Vec::new();
+        let stdout_join_handle = thread::spawn(move || -> io::Result<Option<Vec<u8>>> {
+            let mut collected_stdout = if relay_stdout { None } else { Some(Vec::new()) };
```

Comment:
Currently `relay_stdout` decides whether `stdout` is captured or not. So `relay_stdout == false` means it's captured and `relay_stdout == true` means, it's not. This might change in the future and there may be ways to either both capture and relay `stdout` or to both *not* capture and *not* relay it. But that's not implemented yet, so I think we should only have one boolean in `Config`. I do wonder whether it's more intuitive to have the boolean field be `capture_stdout` instead.